### PR TITLE
upload task status to b2

### PR DESCRIPTION
**Pull request by ShakoHo**
modify related jobs for status file creator
modify runtest and base test for status file creator
@askeing @ypwalter r?

### CODE REVIEW

File: ejenti/jobs/status_json_creator.py [hunk 1]
```diff
@@ -127,7 +127,7 @@ def status_json_creator(**kwargs):
                         job_utc_ts = data_obj["start_ts"]
                         task_name = data_obj["cmd"]
                         for status_obj in data_obj["status"]:
-                            status_code = status_obj["code"]
+                            status_code = status_obj["fn_code"]
```
Comment (askeing) [inline ejenti/jobs/status_json_creator.py:136]: what does `fn_code` mean?
what's the difference between `code` and `fn_code`?

File: ejenti/jobs/status_json_creator.py [hunk 2]
```diff
@@ -156,23 +156,25 @@ def status_json_creator(**kwargs):
...
         for task_with_extname in task_with_extname_list:
             task_name = task_with_extname.split(os.extsep)[0]
             data_obj = {"type": job_name,
                         "start_ts": job_utc_timestamp_string,
                         "cmd": task_name,
                         "status": []}
-            task_code_file_list = [task_file_name for task_file_name in task_file_name_list if task_file_name.split("-")[1].split(os.extsep)[0] == task_name]
+            task_code_file_list = [task_file_name for task_file_name in task_file_name_list if task_file_name.split("-")[-1].split(os.extsep)[0] == task_name]
             for task_code_file_name in task_code_file_list:
+                fn_task_code = "-".join(task_file_name.split("-")[:-1])
```
Comment (askeing) [inline ejenti/jobs/status_json_creator.py]: `task_file_name` should be replaced by `task_code_file_name`.

File: lib/common/statusFileCreator.py [hunk 3]
```diff
@@ -16,22 +16,82 @@ class StatusFileCreator(object):
...
+    STATUS_CODE_DESC_CMD_EXECUTED_SUCCESSFULLY = "Successfully complete"
+
+    STATUS_CODE_GENERAL_CMD_TEMPLATE = {100: "Prepare to execute the command",
+                                        800: "Failed on exeucting the command",
+                                        900: STATUS_CODE_DESC_CMD_EXECUTED_SUCCESSFULLY}
+
     STATUS_CODE_MAPPING = {STATUS_TAG_PULSE: {100: "Get message from pulse queue",
                                               200: "Put message into local queue",
-                                              900: "Successfully complete"},
+                                              900: STATUS_CODE_GENERAL_CMD_TEMPLATE},
```
Comment (askeing) [inline lib/common/statusFileCreator.py]: should be `STATUS_CODE_DESC_CMD_EXECUTED_SUCCESSFULLY`, not `STATUS_CODE_GENERAL_CMD_TEMPLATE` which is a `dict` object.

File: lib/common/statusFileCreator.py [hunk 4]
```diff
@@ -16,22 +16,82 @@ class StatusFileCreator(object):
...
                            STATUS_TAG_PULSE_TRIGGER: {100: "Detect new build, push tasks to Pulse MQ",
                                                       900: "Push all tasks to Pulse MQ, successfully complete"},
```
Comment (askeing) [inline lib/common/statusFileCreator.py]: Should also change to `STATUS_CODE_DESC_CMD_EXECUTED_SUCCESSFULLY`

File: ejenti/jobs/status_json_creator.py (hunk 1 above)
Comment (ShakoHo) [reply ejenti/jobs/status_json_creator.py:136]: fn_code means the real status code in file name ex: 900-1-pulse.json, fn_code would be 900-1 and code would be 900

Comment (askeing) [reply ejenti/jobs/status_json_creator.py:136]: 👍 
How about rename the `fn_code` to other name? or add the comment in `data example` to describe the meaning of `code` and `fn_code`?

File: runtest.py [hunk 5]
```diff
@@ -112,7 +113,14 @@ def __init__(self, **kwargs):
         else:
             self._firefox_profile_path = ""
 
-        # chrome config
+        # init status job id path, if not specify in exec config, default will be empty string
+        status_job_id = self.exec_config.get("status-job-id", "")
+        if status_job_id:
+            self.status_job_id_path = os.path.join(StatusFileCreator.get_status_folder(), self.exec_config.get("status-job-id", ""))
+        else:
```
Comment (ypwalter) [inline runtest.py]: We will get either "" or content from `status_job_id = self.exec_config.get("status-job-id", "")`.
The if-else here can be removed?

**Review by ypwalter — APPROVED** [review]
r+

**Pull request merged**